authentication.md

```diff
@@ -1,30 +0,0 @@
-## With `gcloud-node`
```

Comment:
These are actually versioned, I think they ended up in the root directory due to a build script error that has since been resolved.

https://github.com/GoogleCloudPlatform/gcloud-node/blob/gh-pages/json/master/authentication.md